.wp-env.json

```
@@ -1,11 +1,17 @@
 {
   "core": null,
-  "plugins": [ "." ],
+  "plugins": [
+    ".",
+    "./plugins/auto-sizes",
+    "./plugins/dominant-color-images",
+    "./plugins/speculation-rules",
+    "./plugins/webp-uploads"
+  ],
   "env": {
     "tests": {
       "config": {
         "FS_METHOD": "direct"
       }
     }
   }
-}
+}
```

EOF EOL should be retained, no?

---

Per `.editorconfig`, JSON files should _not_ have a final newline.

---

Why was that added? It's not in https://github.com/WordPress/wordpress-develop/blob/trunk/.editorconfig

---

No good reason I think, other than it has always been like that. The reverse question though: Why _should_ JSON files have a newline at the end?

---

AFAIK because it's standard for text files to end in newlines. For example https://gist.github.com/OleksiyRudenko/d51388345ea55767b7672307fe35adf3 which links to https://stackoverflow.com/a/729795/93579.

See also https://unix.stackexchange.com/a/18789/218144. 

In short, it seems text utilities expect newlines to work properly.

---

Looking at the other .json files in the project, they all mostly have extra lines and it looks like my editor is auto-removing the lines when I save any of these files because of the .editorconfig. I'll open a new PR for us to make the switch, but for consistency will update the file here.

```suggestion
}

```